lib/autoupdate/agent/setup.go

```diff
@@ -72,6 +74,32 @@ func Setup(ctx context.Context, log *slog.Logger, linkDir, dataDir string) error
 	return nil
 }
 
+// Teardown removes all traces of the auto-updater, including its configuration.
+func Teardown(ctx context.Context, log *slog.Logger, linkDir, dataDir string) error {
+	svc := &SystemdService{
+		ServiceName: "teleport-update.timer",
+		Log:         log,
+	}
+	if err := svc.Disable(ctx); err != nil {
+		return trace.Errorf("failed to disable teleport-update systemd timer: %w", err)
```

Comment:
I'm not sure I understand. I'm using `trace.Errorf` specifically because it uses vanilla Go to do the wrapping:
https://github.com/gravitational/trace/blob/master/trace.go#L190

This works exactly like I'd expect it to, and avoids injecting whitespace that I don't want in log messages: https://go.dev/play/p/SWAADglGlJf

Maybe the issue you're describing was resolved here? https://github.com/gravitational/trace/commit/1cff4537bbdf3f4e9f81c6d64cd51bbc56d24bd5

It seems like we would want errors that are wrapped with `fmt.Errorf` inside third-party libraries to be treated the same as errors wrapped in teleport. The formatting `trace.Errorf` applies seems more correct in that respect. Otherwise the log output would be inconsistent depending on where the error was wrapped.

When I've asked about the correct convention before, actually heard we're moving towards `trace.Errorf`, since it more closely matches the idiomatic Go convention. Maybe I misunderstood though -- happy to discuss in more detail. Since the entire package uses `trace.Errorf` to wrap, I'd rather keep this PR consistent and fix everything together in a follow up PR if needed.

Reply:
In the RFD I linked:

> Currently trace.Wrap() implementation is wrapping only once, subsequent wraps only append messages to the existing traced error. Go's standard libraries are consecutively wrapping and creating an error tree that supports both single error wrapping and error joining.

The `trace` package goal is not only to append to the error message but to capture the stacktrace and generate debug reports. By wrapping an already traced error with go-style wrapping, you are breaking the existing trace and starting a new one. `trace.Wrap` will capture again the stacktrace and lose the initial error context.

Demo: https://go.dev/play/p/TQlR6zRZRcI

> When I've asked about the correct convention before, actually heard we're moving towards `trace.Errorf`

This must have been a misunderstanding, or maybe the person who told you that might not be familiar with what the `trace` package does. With the current implementation `trace.Errorf` must not be used to wrap already traced errors, only to create new errors from scratch. I understand that this is confusing, this is the reason I proposed rewriting trace to follow the new go standards. Until the `trace` package is updated, this pattern should be avoided.

Reply:
Ah, thanks, I assumed it would unwrap and append to the trace, but keep the original error clean. I haven't found these error-embedded traces to be especially useful, so I guess I've never noticed that the trace gets lobbed off. I'll follow up on the Cloud side about `trace.Errorf` so we have consistent messaging.

I can follow up with another PR converting the package to Wrap consistently. Is there a way to avoid the tabs and newlines between errors in structured logs, or disable the stack trace without losing the underlying error when I don't want users to see it?

Reply:
https://github.com/gravitational/teleport/pull/49388